specifyweb/frontend/js_src/lib/components/DataModel/__tests__/businessRules.test.ts

```diff
@@ -1,194 +1,144 @@
-      id: 1,
+describe('Borrow Material business rules', () => {
+  const borrowMaterialId = 1;
+  const borrowMaterialUrl = getResourceApiUrl(
+    'BorrowMaterial',
+    borrowMaterialId
+  );
+
+  const getBaseBorrowMaterial = () =>
+    new schema.models.BorrowMaterial.Resource({
+      id: borrowMaterialId,
+      resource_uri: borrowMaterialUrl,
+      quantity: 20,
+      quantityreturned: 13,
+      quantityresolved: 15,
     });
-    dnaSequence.set('geneSequence', 'cat123gaaz');
 
-    expect(dnaSequence.get('totalResidues')).toBe(10);
-    expect(dnaSequence.get('compA')).toBe(3);
-    expect(dnaSequence.get('ambiguousResidues')).toBe(4);
+  test('fieldCheck quantityReturned', async () => {
+    const borrowMaterial = getBaseBorrowMaterial();
+
+    borrowMaterial.set('quantityReturned', 30);
+    expect(borrowMaterial.get('quantityReturned')).toBe(15);
   });
-});
 
-describe('uniquenessRules', () => {
-  const permitOneId = 1;
-  const permitOneUrl = '/api/specify/permit/1/';
-  const permitOneResponse = {
-    id: permitOneId,
-    resource_uri: permitOneUrl,
-    permitNumber: '20',
-  };
-  overrideAjax(permitOneUrl, permitOneResponse);
-
-  const permitTwoId = 2;
-  const permitTwoUrl = getResourceApiUrl('Permit', permitTwoId);
-  const permitTwoResponse = {
-    id: permitTwoId,
-    resource_uri: permitTwoUrl,
-    permitNumber: '20',
-  };
-  overrideAjax(permitTwoUrl, permitTwoResponse);
-
-  overrideAjax(getResourceApiUrl('CollectionObject', 221), {
-    id: 221,
-    resource_uri: getResourceApiUrl('CollectionObject', 221),
-    catalogNumber: '000022002',
+  test('fieldCheck quantityResolved', async () => {
+    const borrowMaterial = getBaseBorrowMaterial();
+
+    borrowMaterial.set('quantityResolved', 30);
+    expect(borrowMaterial.get('quantityResolved')).toBe(20);
+
+    borrowMaterial.set('quantityResolved', 5);
+    expect(borrowMaterial.get('quantityResolved')).toBe(13);
   });
+});
 
-  test('global uniquenessRule', async () => {
-    const testPermit = new schema.models.Permit.Resource({
-      id: permitOneId,
-      permitNumber: '20',
-    });
-    await testPermit.save();
+describe('Collection Object business rules', () => {
+  const collectionObjectlId = 2;
+  const collectionObjectUrl = getResourceApiUrl(
+    'CollectionObject',
+    collectionObjectlId
+  );
 
-    const duplicatePermit = new schema.models.Permit.Resource({
-      id: permitTwoId,
-      permitNumber: '20',
-    });
-    expect(
-      duplicatePermit
-        .fetch()
-        .then((permit) =>
-          permit.businessRuleManager?.checkField('permitNumber')
-        )
-    ).resolves.toBe({
-      key: 'br-uniqueness-permitnumber',
-      valid: false,
-      reason: 'Value must be unique to Database',
+  const getBaseCollectionObject = () =>
+    new schema.models.CollectionObject.Resource({
+      id: collectionObjectlId,
+      resource_uri: collectionObjectUrl,
     });
+
+  beforeEach(() => {
```

Comment:
should use `beforeAll` instead. although that doesn't make a difference in this case, `beforeEach` should be used when an action needs to be repeated for each test (i.e setup or teardown that needs to be reset for each test - possibly because the test might have modified something)

also, don't forget to use `afterAll` to change the value back afterward to original value (or optionally modify the mocked API response in the .json file, and see if it breaks any other tests)

Reply:
Fixed in https://github.com/specify/specify7/pull/3792/commits/7a9d0b79016dad7adeb95f333e41479290935b13